src/rpc/services/JoinChannelService.ts

```diff
@@ -21,68 +37,316 @@ class JoinChannelService extends ARpcService {
         signedJoinChannel: SignedJoinChannelStruct,
         confirmationSignature?: SignatureLike
     ) {
-        //TODO! - require seccusfull init handshake
         try {
-            let key = signedJoinChannel.encodedJoinChannel.toString(); //shouldn't leave a big memory footprint since it will be pruned
-            let jc = EvmUtils.decodeJoinChannel(
+            const key = signedJoinChannel.encodedJoinChannel.toString();
+            const joinChannel = EvmUtils.decodeJoinChannel(
                 signedJoinChannel.encodedJoinChannel
             );
-            if (!this.joinChannelMap.has(key)) {
-                let timeRemaining =
-                    Number(jc.deadlineTimestamp) - Clock.getTimeInSeconds();
-                if (timeRemaining <= 0) return;
-                let retrivedAddress = EvmUtils.retrieveSignerAddressJoinChannel(
-                    jc,
+
+            // Validate request timeframe
+            const timeRemaining =
+                Number(joinChannel.deadlineTimestamp) -
+                Clock.getTimeInSeconds();
+            if (timeRemaining <= 0) {
+                this.joinChannelMap.delete(key);
+                return; // Request expired
+            }
+            const isNewRequest = !this.joinChannelMap.has(key);
+
+            // Handle new request initialization if needed
+            if (isNewRequest) {
+                // Validate the request
+                const validationResult = await this.validateOriginalRequest(
+                    joinChannel,
                     signedJoinChannel.signature as SignatureLike
                 );
-                if (jc.participant != retrivedAddress) return; //TODO! Disconnect from peer
-                //TODO! - analyzie retrivedAddress and apply subjectivivity to decide if to proceed or not
-                //TODO! StateObserver can be triggered for this subjectivity too
-                let mySignedJC = await EvmUtils.signJoinChannel(
-                    jc,
+                if (validationResult !== ValidationFlag.VALID) {
+                    console.warn(
+                        `Invalid original request: ${ValidationFlag[validationResult]}`
+                    );
+                    return;
+                }
+
+                // Initialize the request with original signature
+                this.initializeRequest(
```

Comment:
name change

Reply:
removed function. baked the timeout functionality into the `SignatureCollectionMap`